docs/integrations/sources/quickbooks.md

```diff
@@ -1,12 +1,76 @@
 # QuickBooks
 
-## Overview
+This page contains the setup guide and reference information for the QuickBooks Source connector.
 
-The QuickBooks source supports both Full Refresh and Incremental syncs. You can choose if this connector will copy only the new or updated data, or all rows in the tables and columns you set up for replication, every time a sync is run.
+## Prerequisites
 
-This source wraps the [Singer QuickBooks Tap](https://github.com/singer-io/tap-quickbooks).
+- Account at [Intuit Developer account](https://developer.intuit.com/app/developer/qbo/docs/get-started)
+- Start date
+- Realm ID
+- User Agent
+
+<!-- env:oss -->
+**For Airbyte Open Source:**
+
+- OAuth2.0 credentials (see [OAuth 2.0 playground](https://developer.intuit.com/app/developer/qbo/docs/develop/authentication-and-authorization/oauth-2.0-playground))
+<!-- /env:oss -->
+
+## Setup guide
+
+### Step 1: Set up QuickBooks
+
+Create a [Intuit Developer account](https://developer.intuit.com/app/developer/qbo/docs/get-started)
+
+<!-- env:oss -->
+**Airbyte Open Source additional setup steps**
+
+1. Create an application
+2. Obtain credentials. The easiest way to get these credentials is by using Quickbook's [OAuth 2.0 playground](https://developer.intuit.com/app/developer/qbo/docs/develop/authentication-and-authorization/oauth-2.0-playground)
+
+<!-- /env:oss -->
+
+<!-- env:cloud -->
+### Step 2: Set up the QuickBooks connector in Airbyte
+
+**For Airbyte Cloud:**
+
+1. [Log into your Airbyte Cloud](https://cloud.airbyte.com/workspaces) account.
+2. In the left navigation bar, click **Sources**. In the top-right corner, click **+ new source**.
+3. On the source setup page, select **QuickBooks** from the Source type dropdown and enter a name for this connector.
+4. Click `Authenticate your QuickBooks account`
+5. Log in and Authorize to the QuickBooks account.
+6. **Start date** - The date from which you'd like to replicate data for streams.
+7. **Realm ID** - The Labeled Company ID you'd like to replicate data for streams.
```

Comment:
where do I get this realm ID? Can we get it on behalf of the user?

Reply:
The doc states we can retrieve the realm ID along with the code from the OAuth response (first one).
![image](https://user-images.githubusercontent.com/7703835/230879937-d7fed1c0-5f92-4480-9c94-a969c2d7b0bc.png)

I'll try implementing this approach but it looks like we have never done this before

Reply:
Doesn't this mean we can retrieve it on behalf of the user during the oauth flow? (And therefore this field should be hidden in the UI in cloud)?

Reply:
Done https://github.com/airbytehq/airbyte-platform-internal/pull/5743